data/sql/headers.go

```diff
@@ -154,6 +155,16 @@ const (
 	FROM headers
 	WHERE hash = ?
 	`
+
+	sqlMerkleRootsConfirmationsStartQuery = `
+	WITH merkles(merkleroot) AS (VALUES 
+	`
+	sqlMerkleRootsConfirmationsEndQuery = `
+	) SELECT m.merkleroot AS merkleroot, h.hash AS hash, h.hash is not null AS confirmed
+	FROM merkles m 
+	LEFT JOIN (SELECT hash, header_state, merkleroot FROM headers WHERE header_state = 'LONGEST_CHAIN') h 
+	ON m.merkleroot = h.merkleroot
+	`
```

Comment:
Instead of this subselect just add a where clause at the end

```suggestion
	LEFT JOIN headers h 
	ON m.merkleroot = h.merkleroot
	where h.header_state = 'LONGEST_CHAIN'
	`
```

Reply:
Nope, because this will only return those merkleroots that are in the longest chain headers. Remember, we also want to return those unconfirmed ones from the query, so that we don't have to append them at the end in code.

Reply:
ohhh, you're right,
then slighty different proposition:
```suggestion
	LEFT JOIN headers h 
	ON m.merkleroot = h.merkleroot and h.header_state = 'LONGEST_CHAIN'
	`
```

Reply:
Nice, thanks, changed

Reply:
It really does give the best result, time of my query is now below `1ms`